src/endpoints/ECatCourse/index.ts

```diff
@@ -744,6 +756,190 @@ class ECatCourse extends EndpointCategory {
+      const {
+        oldStart,
+        oldEnd,
+        newStart,
+        newEnd,
+        daySubstitutionMap = {},
+      } = dateShiftOptions;
+
+      const dayNumberSubstitutionMap: { [k: number]: number } = {};
+      Object.keys(daySubstitutionMap).forEach((k) => {
+        dayNumberSubstitutionMap[dayOfWeekToNumber[
+          k as keyof typeof daySubstitutionMap]
+        ] = dayOfWeekToNumber[daySubstitutionMap[k as keyof typeof daySubstitutionMap]];
+      });
+
+      params.date_shift_options = {
+        shift_dates: true,
+        old_start_date: oldStart,
+        old_end_date: oldEnd,
+        new_start_date: newStart,
+        new_end_date: newEnd,
+        day_substitutions: dayNumberSubstitutionMap,
+      };
+    }
+
+    try {
+      const contentMigration = await this.visitEndpoint({
+        path: `${API_PREFIX}/courses/${destinationCourseId}/content_migrations`,
+        action: 'perform a course content migration',
+        method: 'POST',
+        params,
+      });
+
+      let workflowState = 'running';
+      let migrationIssuesCount = 0;
+
+      const CHECK_INTERVAL_MS = 500;
+      const numIterations = Math.ceil(timeoutMs / CHECK_INTERVAL_MS);
+      for (let i = 0; i < numIterations; i++) {
+        await new Promise((resolve) => {
+          setTimeout(resolve, CHECK_INTERVAL_MS);
+        });
+        const status = await this.visitEndpoint({
+          path: `${API_PREFIX}/courses/${destinationCourseId}/content_migrations/${contentMigration.id}`,
+          action: 'check the status of a content migration',
+          method: 'GET',
+        });
+        workflowState = status.workflow_state;
+        migrationIssuesCount = status.migration_issues_count;
+
+        if (workflowState === 'completed' || workflowState === 'failed') {
+          break;
+        }
+      }
+      if (workflowState !== 'completed' && workflowState !== 'failed') {
+        throw new CACCLError({
+          message: 'Migration timed out',
+          code: ErrorCode.MigrationTimeout,
+        });
+      }
+
+      if (migrationIssuesCount > 0) {
+        const migrationIssues = await this.visitEndpoint({
+          path: `${API_PREFIX}/courses/${destinationCourseId}/content_migrations/${contentMigration.id}/migration_issues`,
+          action: 'get migration issues',
+          method: 'GET',
+        });
+
+        let errorMessage = '';
+        if (migrationIssuesCount === 1) {
+          errorMessage = `We ran into an error while migrating your course content: ${migrationIssues[0].description}`;
+        } else {
+          errorMessage = `We ran into ${migrationIssuesCount} errors while migrating your course content: `;
+          for (let i = 0; i < migrationIssues.length; i++) {
+            if (i === migrationIssues.length - 1) {
+              errorMessage += `and ${migrationIssues[i].description}.`;
+            } else {
+              errorMessage += `${migrationIssues[i].description}, `;
```

Comment:
This will read weird if there are two items, but otherwise is pretty good. This is very subtle (extra comma).